libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx

```diff
@@ -138,20 +144,38 @@ export const DateFilterCore: React.FC<IDateFilterCoreProps> = ({
-                            renderBody={({ closeDropdown }) => (
-                                // Dropdown component uses React.Children.map and adds special props to this component
-                                // https://stackoverflow.com/questions/32370994/how-to-pass-props-to-this-props-children
-                                <DateFilterBody
-                                    {...dropdownBodyProps}
-                                    filterOptions={filteredFilterOptions}
-                                    isMobile={isMobile}
-                                    closeDropdown={closeDropdown}
-                                    dateFilterButton={dateFilterButton()}
-                                    dateFormat={verifiedDateFormat}
-                                    isTimeForAbsoluteRangeEnabled={isTimeForAbsoluteRangeEnabled}
-                                    weekStart={weekStart}
-                                />
-                            )}
+                            renderBody={({ closeDropdown }) => {
+                                return isConfigurationOpen ? (
+                                    <FilterConfigurationComponent
+                                        onSaveButtonClick={() => {
+                                            setIsConfigurationOpen(false);
+                                        }}
+                                        onCancelButtonClick={() => {
+                                            setIsConfigurationOpen(false);
+                                            onCancelClick();
+                                        }}
+                                    />
+                                ) : (
+                                    // Dropdown component uses React.Children.map and adds special props to this component
+                                    // https://stackoverflow.com/questions/32370994/how-to-pass-props-to-this-props-children
+                                    <DateFilterBody
+                                        {...dropdownBodyProps}
+                                        onApplyClick={onApplyClick}
+                                        onCancelClick={onCancelClick}
+                                        filterOptions={filteredFilterOptions}
+                                        isMobile={isMobile}
+                                        closeDropdown={closeDropdown}
+                                        dateFilterButton={dateFilterButton()}
+                                        dateFormat={verifiedDateFormat}
+                                        isTimeForAbsoluteRangeEnabled={isTimeForAbsoluteRangeEnabled}
+                                        weekStart={weekStart}
+                                        isConfigurationEnabled={FilterConfigurationComponent !== undefined}
+                                        onConfigurationClick={() => {
+                                            setIsConfigurationOpen(true);
```

Comment:
The inline function makes the component be rerenderred, please create a separate function for this prop!